WcaOnRails/app/models/competition.rb

```diff
@@ -266,11 +294,37 @@ def registration_must_close_after_it_opens
     end
   end
 
+  def use_wca_as_website=(value)
+    @use_wca_as_website = ActiveRecord::Type::Boolean.new.type_cast_from_database(value)
+  end
+
+  def use_wca_as_website
+    @use_wca_as_website || website == url
+  end
+
+  before_validation :assign_wca_website_url
+  private def assign_wca_website_url
+    if use_wca_as_website && website != url
+      self.website = self.url
+    end
+  end
+
+  after_save :update_website
+  private def update_website
+    if use_wca_as_website && website != url
+      update_column :website, url
```

Comment:
Reusing the model as an object for the form (even though the attributes don't match the form fields) is what leads to this kind of indirection. I'm not here to complain; there are more examples in other models and it obviously works. An _ugly_ `if params[: use_wca_as_website]` in the controller would work as well, though.